classes/Configuration.php

```diff
@@ -601,13 +613,26 @@ public static function deleteFromContext($key, int $idShopGroup = null, int $idS
             $idShop = Shop::getContextShopID(true);
         }
 
-        $id = Configuration::getIdByName($key, $idShopGroup, $idShop);
+        $configurationId = Configuration::getIdByName($key, $idShopGroup, $idShop);
+
+        self::deleteFromGivenContext($key, $configurationId, $idShopGroup, $idShop);
+    }
+
+    /**
+     * @param mixed $key
+     * @param int|null $configurationId
+     * @param int|null $idShopGroup
+     * @param int|null $idShop
+     */
+    public static function deleteFromGivenContext($key, int $configurationId = null, int $idShopGroup = null, int $idShop = null)
```

Comment:
I splitted `deleteFromContext` in two, by adding `deleteFromGivenContext`.

So now `deleteFromContext` calls `deleteFromGivenContext` for the SQL part. I did that so that I have a method using shop and shopGroup ids from given context parameters only (and not from the current context)

Reply:
```suggestion
    public static function deleteMultiShopConfigurationByKey($key, ?int $idShopGroup, ?int $idShop)
```

If the purpose is to have a stricter function then the parameters shouldn't be optional, the naming can be improved as well so we understand the actual purpose

Reply:
You could also have a `deleteMultiShopConfigurationById` (maybe it could be protected since it might only be used internally)

Reply:
```suggestion
    public static function deleteFromGivenContext($key, int $configurationId = null, int $idShopGroup = null, int $idShop = null): void
```